### PR TITLE
[Story] Transform project sub pages into one page with sections

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: frontend/components/elements/TopicItem.vue [hunk 1]
```diff
@@ -4,15 +4,23 @@
             <template v-if="!editing">
                 <BButton
                     size="sm"
-                    variant="danger"
+                    variant="outline-danger"
                     class="float-right ml-2"
-                    v-confirm="{ action: deleteTopic, text: 'Are you sure you want to delete this topic?' }">Delete</BButton>
+                    v-confirm="{ action: deleteTopic, text: 'Are you sure you want to delete this topic?' }">
```
Comment (anjakammer) [inline frontend/components/elements/TopicItem.vue]: It is not possible to delete a topic, which relates to stories. 
It comes a 500 error from the api. We should make a proper exception out of it with the message, that it is not possible to delete topics, which are in use.

Currently, the frontend says nothing - the topic just remains without an error.

File: frontend/components/elements/StoryItem.vue [hunk 2]
```diff
@@ -13,45 +13,45 @@
...
-            <td v-if="isView(['backlog', 'planning-sprint', 'planning-backlog'])">
+            <td v-if="isView(['backlog', 'planning-sprint', 'planning-backlog'])" :class="$style.parent">
                 <nobr>
                     <BButton
-                            v-if="isView('planning-backlog')"
-                            size="sm"
-                            variant="primary"
-                            title="Add to sprint"
-                            @click="addToSprint">
+                        v-if="isView('planning-backlog')"
```
Comment (anjakammer) [inline frontend/components/elements/StoryItem.vue:19]: should have another color than the 'remove from sprint' button to get the difference between them.

File: frontend/components/elements/StoryItem.vue [hunk 3]
```diff
@@ -124,42 +123,34 @@
...
             </td>
 
-            <td v-if="isView('backlog')">
+            <td v-if="isView(['backlog', 'planning-backlog', 'planning-sprint'])">
                 <BButton 
                     size="sm" 
-                    variant="primary" 
-                    title="Edit" 
+                    variant="outline-primary"
+                    title="Edit"
```
Comment (anjakammer) [inline frontend/components/elements/StoryItem.vue:130]: The edit button should have the same size as the delete button.

File: frontend/components/elements/StoryItem.vue [hunk 4]
```diff
@@ -124,42 +123,34 @@
...
                     <VIcon name="pencil"/>
+                    Edit
                 </BButton>
 
                 <BButton 
                     size="sm" 
-                    variant="danger" 
+                    variant="outline-danger"
                     title="Delete" 
                     v-confirm="{ action: remove, text: 'Are you sure you want to delete this story?' }">
```
Comment (anjakammer) [inline frontend/components/elements/StoryItem.vue:142]: There is no 7x confirmation loop for deleting a story anymore. :+1:

**Review by anjakammer — COMMENTED** [review]
Thanks for the changes, it looks awesome. I had just some small change requests.

File: frontend/components/elements/StoryItem.vue (hunk 3 above)
Comment (ghost) [reply frontend/components/elements/StoryItem.vue:130]: they're the same size for me? unless you mean the width, which is dependent on the width of the icon + text, and I can't really make the word edit longer...

File: frontend/components/elements/StoryItem.vue (hunk 2 above)
Comment (ghost) [reply frontend/components/elements/StoryItem.vue:19]: I don't think that makes sense - they belong to the same category of button (determining whether the story is in the sprint or not), and they're already differentiated by icon and text - changing the colour as well would just disassociate their function from each other, which is not what we want.

File: frontend/components/elements/StoryItem.vue (hunk 4 above)
Comment (ghost) [reply frontend/components/elements/StoryItem.vue:142]: whoop!

File: frontend/components/elements/StoryItem.vue (hunk 2 above)
Comment (anjakammer) [reply frontend/components/elements/StoryItem.vue:19]: ok

File: frontend/components/elements/StoryItem.vue (hunk 3 above)
Comment (anjakammer) [reply frontend/components/elements/StoryItem.vue:130]: You could specify the width to about 75px for each button of them:
image


It looks much more cleaner.

File: frontend/components/elements/TopicItem.vue [hunk 5]
```diff
@@ -6,7 +6,10 @@
                     size="sm"
                     variant="outline-danger"
                     class="float-right ml-2"
-                    v-confirm="{ action: deleteTopic, text: 'Are you sure you want to delete this topic?' }">
+                    v-confirm="{
+                        action: deleteTopic,
+                        text: 'Are you sure you want to delete this topic? This will remove the topic link from all associated stories.'
```
Comment (anjakammer) [inline frontend/components/elements/TopicItem.vue]: @mvuajua will you implement to removal of the topic references in this PR?

File: frontend/components/elements/StoryItem.vue (hunk 3 above)
Comment (ghost) [reply frontend/components/elements/StoryItem.vue:130]: I actually think the opposite - seeing as icons and text are not aligned anymore, it looks a lot messier to me. if we'd make the buttons fixed width, they should all be at least a bit bigger than the widest one (the delete one in this case), with the icon aligned left and the text center in the remaining space, to create a unifying look - however seeing as space in that table is already limited, I would not want to make the delete button any wider, especially seeing as we don't win anything imo - if we're focusing on this, there's a lot more issues with that page we should look at first...

File: frontend/components/elements/TopicItem.vue (hunk 5 above)
Comment (ghost) [reply frontend/components/elements/TopicItem.vue]: sure! added it.

File: frontend/components/elements/StoryItem.vue (hunk 3 above)
Comment (anjakammer) [reply frontend/components/elements/StoryItem.vue:130]: ok :)

File: app/models/topic.rb [hunk 6]
```diff
@@ -1,6 +1,6 @@
 class Topic < ApplicationRecord
   belongs_to :course
-  has_many :stories
+  has_many :stories, dependent: :nullify
```
Comment (anjakammer) [inline app/models/topic.rb:3]: that was quick :D

**Review by anjakammer — APPROVED** [review]
works perfectly!

**Pull request merged**